packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/InAppPurchasesModule.java

```diff
@@ -0,0 +1,96 @@
+package expo.modules.inapppurchases;
+
+import java.util.List;
+
+import android.content.Context;
+import android.app.Activity;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.services.EventEmitter;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
+
+public class InAppPurchasesModule extends ExportedModule implements ModuleRegistryConsumer {
+  private static final String TAG = InAppPurchasesModule.class.getSimpleName();
+  private static final String NAME = "ExpoInAppPurchases";
+
+  private BillingManager mBillingManager;
+  private ModuleRegistry mModuleRegistry;
+  protected static EventEmitter sEventEmitter;
+
+  public InAppPurchasesModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
+  }
+
+  @ExpoMethod
+  public void connectAsync(final Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity == null) {
+      promise.reject("E_ACTIVITY_UNAVAILABLE", "Activity is not available");
+    }
+    sEventEmitter = mModuleRegistry.getModule(EventEmitter.class);
+    mBillingManager = new BillingManager(activity);
```

Comment:
Pass the event emitter into the BillingManager instead of using the static, and then you can thread it through to the UpdatesListener.